pam/src/conv.rs

```diff
@@ -54,18 +55,21 @@ impl PamConv {
     /// styles.
     pub fn send(&self, style: PamMessageStyle, msg: &str) -> PamResult<Option<String>> {
         let mut resp_ptr: *const PamResponse = ptr::null();
+        let msg_cstr = CString::new(msg).unwrap();
         let msg = PamMessage {
             msg_style: style,
-            msg: CString::new(msg).unwrap().as_ptr(),
+            msg: msg_cstr.as_ptr(),
         };
 
         let ret = (self.conv)(1, &&msg, &mut resp_ptr, self.appdata_ptr);
 
         if PamResultCode::PAM_SUCCESS == ret {
-            if resp_ptr.is_null() {
+            // PamResponse.resp is null for styles that don't return user input like PAM_TEXT_INFO
+            let response = unsafe { (*resp_ptr).resp };
```

Comment:
Not so sure what the correct thing to do is here. From the code it looks like `resp_ptr` should never be null. Should we assume the C code is correct and dereference this no matter what, or add another check to see if `resp_ptr` is null, and throw a `PAM_SYSTEM_ERROR` in that case?